src/components/CurrencyInputPanel/CurrencyBalance.tsx

```diff
@@ -39,20 +42,58 @@ export default function CurrencyBalance({
   onMax,
   hideBalance,
   showMaxButton,
+  networkFees,
+  subentryCount,
 }: CurrencyBalanceProps) {
+  // const [fee, setFee] = useState<number>(0);
+  // const [adjustedBalance, setAdjustedBalance] = useState<number>(0);
   const { tokenBalancesResponse } = useGetMyBalances();
   const balance =
     tokenBalancesResponse?.balances?.find((b) => b?.contract === currency?.contract)?.balance ||
     '0';
 
+  let availableBalance: number;
+  if (currency.code === 'XLM') {
```

Comment:
Should use the XLM contract and not CODE (there could be another asset with the same code)

should use xlmTokenList.find()